src/app/onboarding/availability/page.tsx

```diff
@@ -1,109 +1,213 @@
-      hpm !== undefined &&
-      hpm >= 0 &&
-      onboarding?.userProfile.start_date &&
-      isValidDate(onboarding?.userProfile.start_date)
-    ) {
-      onboarding?.setCanContinue(true);
-    } else {
-      onboarding?.setProgress(2);
-      onboarding?.setCanContinue(false);
-    }
-  }, [onboarding]);
-
-  const handleHours = (v: string) => {
-    const n = parseInt(v, 10);
-    if (Number.isNaN(n)) {
-      onboarding?.updateProfile({
-        hours_per_month: undefined,
-      });
-    } else {
-      onboarding?.updateProfile({
-        hours_per_month: n,
-      });
-    }
-  };
-
-  const handleStartDate = (v: string) => {
-    if (v === '' || isValidDate(v)) {
-      onboarding?.updateProfile({
-        start_date: v,
-      });
-    }
-  };
-
-  const handleAvailability = (v: string) => {
-    onboarding?.updateProfile({
-      availability_description: v,
+    setOnboardingForm({
+      trigger: form.handleSubmit(identity),
+      isDirty,
+      isValid,
     });
+  }, [setOnboardingForm, form, isDirty, isValid]);
+
+  // handle valid form submission
+  // validity should have already been handled by Zod
+  const onValidSubmit = () => {
+    push(`/onboarding/${onboarding.flow[3].url}`);
+    onboarding.setForm(undefined);
   };
 
   return (
-    <>
-      <H1>Availability</H1>
-      <TextInput
-        label="How much time do you have to commit? (hrs/month)"
-        placeholder="hours/month"
-        errorText={getHoursErrorText}
-        type="number"
-        id="hours"
-        value={hours}
-        setValue={setHours}
-        onChange={handleHours}
-      />
-      <DateInput
-        label="What is the earliest you are available to volunteer?"
-        id="start_date"
-        error={getStartDateErrorText}
-        value={startDate}
-        min={parseDateAlt(new Date())}
-        setValue={setStartDate}
-        onChange={handleStartDate}
-      />
-      <TextAreaInput
-        label="Are there specific time periods you will not be available? (Optional)"
-        placeholder="I won't be available from..."
-        id="periods"
-        value={periods}
-        setValue={setPeriods}
-        onChange={handleAvailability}
-      />
-    </>
+    <FormProvider {...form}>
+      {/* noValidate to prevent default HTML invalid input pop-up */}
+      <CardForm onSubmit={form.handleSubmit(onValidSubmit)} noValidate>
+        <Styles.BackLinkButton
+          type="button"
+          onClick={() => ebbTo(backlinkHref)}
+        >
+          <Icon type="leftArrow" />
+        </Styles.BackLinkButton>
+
+        <Styles.RequiredText>Required Fields</Styles.RequiredText>
+
+        <H1Centered>Availability</H1Centered>
+
+        <Styles.FormFieldsContainer>
+          <FormField
+            control={form.control}
+            name="hoursPerMonth"
+            render={({ field, fieldState }) => (
+              <FormItem>
+                <FormLabel>
+                  How many hours are you able to dedicate per month?
```

Comment:
Let's change "dedicate" to "volunteer" here to be more clear and consistent with the second question.